language/move-vm/runtime/src/interpreter.rs

```diff
@@ -56,14 +56,16 @@ macro_rules! set_err_info {
 ///
 /// An `Interpreter` instance is a stand alone execution context for a function.
 /// It mimics execution on a single thread, with an call stack and an operand stack.
-pub(crate) struct Interpreter {
+pub(crate) struct Interpreter<L: Logger> {
     /// Operand stack, where Move `Value`s are stored for stack operations.
     operand_stack: Stack,
     /// The stack of active functions.
     call_stack: CallStack,
+    // Logger to report information to clients
```

Comment:
it makes it easier and more sane to save the logger in the `Interpreter` and use it from here. We would have to pass it around all over otherwise